WalletWasabi.Fluent/ViewModels/Wallets/Home/Tiles/PrivacyRing/PrivacyRingViewModel.cs

```diff
@@ -39,53 +40,60 @@ public PrivacyRingViewModel(WalletViewModel walletViewModel)
 
 		PreviewItems.Add(PrivacyTile);
 
-		_itemsSourceList
+		SetupCancel(enableCancel: true, enableCancelOnEscape: true, enableCancelOnPressed: true);
+	}
+
+	public PrivacyControlTileViewModel PrivacyTile { get; }
+
+	public ObservableCollectionExtended<PrivacyRingItemViewModel> Items { get; } = new();
+	public ObservableCollectionExtended<IPrivacyRingPreviewItem> PreviewItems { get; } = new();
+
+	public Wallet Wallet { get; }
+
+	[System.Diagnostics.CodeAnalysis.SuppressMessage("Reliability", "CA2000:Dispose objects before losing scope", Justification = "Uses DisposeWith()")]
+	protected override void OnNavigatedTo(bool isInHistory, CompositeDisposable disposables)
+	{
+		var itemsSourceList = new SourceList<PrivacyRingItemViewModel>();
+
+		itemsSourceList
 			.Connect()
 			.ObserveOn(RxApp.MainThreadScheduler)
 			.Bind(Items)
 			.DisposeMany()
-			.Subscribe();
-
-		_coinsUpdated =
-			walletViewModel.UiTriggers.PrivacyProgressUpdateTrigger
-						  .Merge(walletViewModel
-						  .WhenAnyValue(w => w.IsCoinJoining)
-						  .ToSignal());
+			.Subscribe()
+			.DisposeWith(disposables);
 
-		_coinsUpdated
-			.Select(_ => walletViewModel.Wallet.GetPockets())
-			.ObserveOn(RxApp.MainThreadScheduler)
-			.Subscribe(RefreshCoinsList);
-
-		_disposables.Add(_itemsSourceList);
+		_walletViewModel.UiTriggers
+						.PrivacyProgressUpdateTrigger
+						.Select(_ => _walletViewModel.Wallet.GetPockets())
+						.ObserveOn(RxApp.MainThreadScheduler)
+						.Subscribe(_ => RefreshCoinsList(itemsSourceList))
+						.DisposeWith(disposables);
 
 		this.WhenAnyValue(x => x.Width, x => x.Height)
+			.Where(tuple => tuple.Item1 != 0 && tuple.Item2 != 0)
+			.Throttle(TimeSpan.FromMilliseconds(100))
+			.ObserveOn(RxApp.MainThreadScheduler)
 			.Subscribe(x =>
 			{
 				var usableHeight = x.Item2;
 				var usableWidth = x.Item1;
 				Margin = new Thickness(usableWidth / 2, usableHeight / 2, 0, 0);
 				NegativeMargin = new Thickness(Margin.Left * -1, Margin.Top * -1, 0, 0);
-				RefreshCoinsList(walletViewModel.Wallet.GetPockets());
-			});
-
-		SetupCancel(enableCancel: true, enableCancelOnEscape: true, enableCancelOnPressed: true);
+				RefreshCoinsList(itemsSourceList);
+			})
+			.DisposeWith(disposables);
```

Comment:
It still renders twice, just not noticeable. You have to merge the two triggers to make it one.

```c#
		this.WhenAnyValue(x => x.Width, x => x.Height)
			.Merge(_walletViewModel.UiTriggers.PrivacyProgressUpdateTrigger.Select(_ => (Width, Height)))
			.Where(tuple => tuple.Item1 != 0 && tuple.Item2 != 0)
			.Throttle(TimeSpan.FromMilliseconds(100))
			.ObserveOn(RxApp.MainThreadScheduler)
			.Subscribe(x =>
			{
				var usableHeight = x.Item2;
				var usableWidth = x.Item1;
				Margin = new Thickness(usableWidth / 2, usableHeight / 2, 0, 0);
				NegativeMargin = new Thickness(Margin.Left * -1, Margin.Top * -1, 0, 0);
				RefreshCoinsList(itemsSourceList);
			})
			.DisposeWith(disposables);
```

Reply:
Or you can use `CombineLatest` instead of Merge, probably nicer.